pliers/extractors/audio.py

```diff
@@ -488,4 +495,126 @@ class PercussiveExtractor(LibrosaFeatureExtractor):
     For details on argument specification visit:
     https://librosa.github.io/librosa/effect.html.'''
 
+
     _feature = 'percussive'
+
+
+class AudiosetLabelExtractor(AudioExtractor):
+    
+    ''' Extract probability of 521 audio event classes based on AudioSet
+    corpus using a YAMNet architecture. Code available at:
+    https://github.com/tensorflow/models/tree/master/research/audioset/yamnet 
+
+    Args:
+    hop_size (float): size of the audio segment (in seconds) on which label 
+        extraction is performed.
+    top_n (int): specifies how many of the highest label probabilities are 
+        returned. If None, all labels (or all in labels) are returned.
+        Top_n and labels are mutually exclusive arguments.
+    labels (list): specifies subset of labels for which probabilities 
+        are to be returned. If None, all labels (or top_n) are returned.
+        The full list of labels is available in the audioset/yamnet 
+        repository (see yamnet_class_map.csv).
+    weights_path (optional): full path to model weights file. If not provided,
+        weights from pretrained YAMNet module are used.
+    yamnet_kwargs (optional): Optional named arguments that modify input 
+        parameters for the model (see params.py file in yamnet repository)
+    '''
+
+    _log_attributes = ('hop_size', 'top_n', 'labels', 'weights_path',
+                       'yamnet_kwargs')
+
+    def __init__(self, hop_size=0.1, top_n=None, labels=None,
+                 weights_path=None, yamnet_path=None, **yamnet_kwargs):
+        if yamnet_path is None:
+            yamnet_path = YAMNET_PATH
+        try:
+            sys.path.insert(0, str(yamnet_path))
+            self.yamnet = attempt_to_import('yamnet')
+            verify_dependencies(['yamnet'])
+        except MissingDependencyError: 
```

Comment:
Why not just automatically install it for the user if it's missing?

Reply:
I think that's a little heavy handed and inconsistent with how pliers handles other `MissingDependency` errors, such as the extra downloads needed for librosa.

Reply:
> I think that's a little heavy handed and inconsistent with how pliers handles other `MissingDependency` errors, such as the extra downloads needed for librosa.
Yep, throwing a missing dependency error is consistent with what we do also, e.g., for nltk. I'd  lean towards letting the user trigger the download (I guess that is the point of having a `MissingDependencyError` in general), since it's also 125Mb of stuff. But no big deal to edit this so to do it for them, if you have a strong take on it.